Documentation/ApiOverview/FlexForms/Index.rst

```diff
@@ -0,0 +1,317 @@
+* The `bootstrap_package <https://extensions.typo3.org/extension/example/>`__
+  uses FlexForms to configure rendering options,
+  e.g. a transition interval and transition type (slide, fade)
+  for the carousel.
+
+.. image:: Images/FlexFormCarousel.png
+   :class: with-shadow
+
+
+How it Works
+============
+
+#. In the extension, a configuration schema is defined and attached to one or more plugins.
+#. When the plugin is added to a page, it can be configured as defined by the configuration
+   schema.
+#. The configuration for this content element is automatically saved to `tt_content.pi_flexform`.
+#. The extension can read current configuration and act according to
+   the configuration.
+
+Steps to Perform (Extension Developer)
+======================================
+
+.. rst-class:: bignums-xxl
+
+#. Create configuration schema in :ref:`T3DataStructure <t3ds>` format (XML)
+
+   Example: :file:`Configuration/FlexForms/Registration.xml`.
+
+   .. code-block:: xml
+
+        <?xml version="1.0" encoding="utf-8" standalone="yes" ?>
+        <T3DataStructure>
+            <sheets>
+                <sDEF>
+                    <ROOT>
+                        <TCEforms>
+                            <sheetTitle>LLL:EXT:extkey/Resources/Private/Language/Backend.xlf:settings.registration.title</sheetTitle>
+                        </TCEforms>
+                        <type>array</type>
+                        <el>
+                            <!-- Add settings here ... -->
+
+                            <!-- Example setting: input field with name settings.timeRestriction -->
+                            <settings.timeRestriction>
+                                <TCEforms>
+                                    <label>LLL:EXT:example/Resources/Private/Language/Backend.xlf:settings.registration.timeRestriction</label>
+                                    <config>
+                                        <default></default>
+                                        <type>input</type>
+                                        <size>15</size>
+                                    </config>
+                                </TCEforms>
+                            </settings.timeRestriction>
+
+                            <!-- end of settings -->
+
+                        </el>
+                    </ROOT>
+                </sDEF>
+            </sheets>
+        </T3DataStructure>
+
+
+#. The configuration schema is attached to one or more plugins
+
+   The vendor name is **Myvendor**, the extension key is **example**
+   and the plugin name is **Registration**.
+
+   In :file:`Configuration/TCA/Overrides/tt_content.php` add the following:
+
+   .. code-block:: php
+
+       $GLOBALS['TCA']['tt_content']['types']['list']['subtypes_addlist']['example_registration'] = 'pi_flexform';
+       \TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addPiFlexFormValue(
+           // plugin signature: <extension key> '_' <plugin name in lowercase>
```

Comment:
`extension key` might be something like `vendor_name` that's typical for agencies, e.g. `cdx_site`, in that case the signature would be `cdxsite_registration`. The example is therefore not covering underscores within extension keys, we might add this one. Which you already mention within the tip.

I would have expected it the information here: https://docs.typo3.org/m/typo3/reference-coreapi/master/en-us/ExtensionArchitecture/NamingConventions/Index.html which you already reference below within tip. But I couldn't find the information there. We should add it there.